### ISSUE TITLE
Switch license to GPL-3.0

### ISSUE THREAD

**Issue by leviem1** [walterhiggins/ScriptCraft#449]
The issue no plugin developer wants to hear: "I think your project is licensed incorrectly".

Because Bukkit is GPL, so is anything linking to it, making this project technically GPL. Luckily, because all contributions were made under MIT, switching to GPL won't require any other contributor's consent.

Though Bukkit *has* mentioned that they don't intend to take legal action for this, this project *is technically* in violation of copyright. I'll leave this up to you!

**Comment by samkass** [comment]
That is not technically true. Copyright only applies to copying. ScriptCraft can use any API it wants and still not violate copyright as long as it doesn't reproduce or distribute the original work. It would be the end-user who downloads it who would "link" it and be bound by any license terms of both parts. To say otherwise would go beyond even what Oracle was trying to argue about use of APIs.

Looking at the ScriptCraft git, I think there is a risk that putting the "spigot-api" jar in the /lib directory is the only questionable part-- if that were changed so the build script downloaded it from upstream, that should solve any issue, if the copyright holder wanted to make a big deal out of it. The rest is fine and can use whatever license it wants, in my humble opinion.

(I am not a lawyer.)

**Comment by leviem1** [comment]
I totally agree that this has no legal weight, though it's been discussed rather widely in the Bukkit community for years now.

I think it comes down to answering "[When is a program and its plug-ins considered a single combined program?](http://www.gnu.org/licenses/gpl-faq.html#GPLPlugins)". I don't think there's any real precedence for this and hopefully there won't be, but regardless it's questionable.

I'm not a lawyer either. I only bring this up because another developer warned me about this recently and it's been questioned by other developers before.

**Comment by samkass** [comment]
But that's not actually the question. The question is, is ScriptCraft doing anything that requires the use of Bukkit's license at all? They're not distributing Bukkit. They're not distributing a larger program that contains Bukkit. If Bukkit tries to argue that simply using their API constitutes Copyright violation they would be on unprecedented legal ground. I think ScriptCraft is safe.

**Comment by leviem1** [comment]
True, though you're not a lawyer and neither am I. Here's my "evidence" though.

"[Linking a GPL covered work statically or dynamically with other modules is making a combined work based on the GPL covered work...](https://www.gnu.org/licenses/gpl-faq.en.html#GPLStaticVsDynamic)"

"[If the main program and the plugins are a single combined program then this means you must license the plug-in under the GPL or a GPL-compatible free software license and distribute it with source code in a GPL-compliant way. A main program that is separate from its plug-ins makes no requirements for the plug-ins.](https://www.gnu.org/licenses/gpl-faq.en.html#GPLAndPlugins)"

"[Before you say anything we are aware that using the GPL will force plugin authors to GPL their plugin too...](https://bukkit.org/threads/what-license-is-the-bukkit-project-under.154/)"

I'm going to leave this conversation here since this has become a discussion between armchair lawyers and should be carefully considered by the copyright holders, not by us.

**Comment by samkass** [comment]
ScriptCraft isn't linking a GPL covered work, though, so the rest is irrelevant. The person who downloads the plugin and uses it is. The ONLY thing Copyright protects is the right to copy, so if you're not doing that, the license is irrelevant because you don't need one. Whoever wrote the blue text above seems to be confusing the distribution of a plug-in with the distribution of an entire combined work.

**Comment by leviem1** [comment]
```java
import org.bukkit.plugin.java.JavaPlugin;
public class ScriptCraftPlugin extends JavaPlugin
```

That seems like a dynamic link to a GPL covered work that is required to make the plugin run. Which essentially means that ScriptCraft is inseparable from and a derived work of Bukkit. Pretty much saying that plugins are a thin layer around a GPL work.

The quotes above are from the Mojang-hired Bukkit staff member EvilSeph (Mojang owns Bukkit) and the Free Software Foundation.

I'd say better safe than sorry.

Okay, now I'm done! 😄

**Comment by samkass** [comment]
That's an import statement-- it's **not** _copying_ the original work. I'm guessing EvilSeph isn't a US lawyer either, or he'd know better. He's just plain wrong. I think you should stop raising incorrect and pointless licensing issues in a drive-by manner, causing trouble, then running away. It's clear that if you're not copying anything, then Copyright law doesn't apply. If someone were to release a distribution that linked ScriptCraft into GPL code, they would be liable-- that's how Copyright (at least in the US) works. But not if you just release your own original work. Maybe it's different in Sweden?

In short, no one should worry about it unless you're sent a cease and desist letter (which has to specify what work you made an illegal copy of), and even then, the right thing is to fight this nonsense. Can we close this pointless thread?
